### ISSUE TITLE
[Feature]:  Add UUID in backup path

### ISSUE THREAD

**Issue by epollia** [cloudnative-pg/cloudnative-pg#3521]
### Is there an existing issue already for this feature request/idea?

- [X] I have searched for an existing issue, and could not find anything. I believe this is a new feature request to be evaluated.

### What problem is this feature going to solve? Why should it be added?

When recreating a cluster with existing backup files on S3, the new cluster cannot create backup.
For example:

1.   We have a postgres cluster with backups on S3.
2.   The cluster was deleted, but the backups on s3 were not cleared.
3.   Create a cluster of the same variables (kubectl apply -f cluster.yalm)
4.   Backup from new cluster failed because some files exist on s3.

### Describe the solution you'd like

The Zolando operator uses the cluster uuid and sets it to the backup path, and the new cluster uses a different backup path. Is it possible to add variable to automatically set the uuid to the backup path?

### Describe alternatives you've considered

We use helm chart and create custom config map with UUID. UUID generated by helm install command, and read by helm upgrade.
This uuid we set in backup path.

### Backport?

Yes

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by gbartolini** [comment]
Have you looked into the `.spec.backup.barmanObjectoStore. serverName` option? see https://cloudnative-pg.io/documentation/current/cloudnative-pg.v1/#postgresql-cnpg-io-v1-BarmanObjectStoreConfiguration

**Comment by epollia** [comment]
Yes, but .spec.backup.barmanObjectoStore.serverName needs to be configured manually. We would like the operator to be able to use the cluster UUID in place of the cluster name.
